routesrv/eskipbytes.go

```diff
@@ -50,6 +59,10 @@ func (e *eskipBytes) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	span := tracing.CreateSpan("serve_routes", r.Context(), e.tracer)
 	defer span.Finish()
 
+	if e.useEtag && e.hash != "" && r.Header.Get("Etag") == e.hash {
```

Comment:
I think this should check `If-None-Match` header, lets verify with the spec https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag

Reply:
Lets check if can utilize https://pkg.go.dev/net/http#ServeContent as it should implement proper support of caching headers